File: app/src/main/java/com/example/listenapp/NewAccountActivity.java
package com.example.listenapp;

import android.os.Bundle;
import android.support.design.widget.CoordinatorLayout;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;


public class NewAccountActivity extends AppCompatActivity {

    EditText user, password, passwordRepeat;
    Button confirm;
    Bundle accInfo;
    CoordinatorLayout layout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        findViews();
        bundleSetup();
        setClicks();
    }
    private void findViews() {
        setContentView(R.layout.activity_new_account);
        user = findViewById(R.id.input_user);
        password = findViewById(R.id.input_pw);
        passwordRepeat = findViewById(R.id.input_pwrepeat);
        confirm = findViewById(R.id.confirm_Button);
        layout = findViewById(R.id.coordinator);
    }
    private void bundleSetup() {
        accInfo = new Bundle();

        //putString("chave", "Não, aí eu coloco a variavel");

    }
    private void setClicks() {
        confirm.setOnClickListener(goTO(AccPlaceholder.class, accInfo));

    }
    private View.OnClickListener goTO(final Class target, final Bundle targetBundle) {
        return new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                if(user.getText().toString().isEmpty() || password.getText().toString().isEmpty() || passwordRepeat.getText().toString().isEmpty()){
                    new CustomSnackbar().makeSB(layout, getString(R.string.empty_fields_msg),15,15).show();
                }
                else {

                }
                //Activity activity = (Activity) view.getContext();
                //activity.startActivity(new Intent(activity, target).putExtras(targetBundle));
            }
        };
    }
}

File: app/src/main/java/com/example/listenapp/main/NewAccountActivity.kt
package com.example.listenapp.main

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import androidx.coordinatorlayout.widget.CoordinatorLayout
import androidx.lifecycle.Observer
import com.example.listenapp.R
import com.example.listenapp.viewmodel.NewAccountViewModelViewModel
import com.google.firebase.auth.FirebaseAuth

class NewAccountActivity : AppCompatActivity() {
    lateinit var user: EditText
    lateinit var password: EditText
    lateinit var passwordRepeat: EditText
    lateinit var confirm: Button
    lateinit var login: Button
    lateinit var accInfo: Bundle
    lateinit var layout: CoordinatorLayout
    lateinit var loginFirebaseAuth: FirebaseAuth
    lateinit var authStateListener: FirebaseAuth.AuthStateListener
    lateinit var firebaseAuth: FirebaseAuth

    private val viewModel : NewAccountViewModelViewModel by viewModels()


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_new_account)
        user = findViewById(R.id.input_user)
        password = findViewById(R.id.input_pw)
        passwordRepeat = findViewById(R.id.input_pwrepeat)
        confirm = findViewById(R.id.confirm_Button)
        login = findViewById(R.id.toLogin_Button)
        layout = findViewById(R.id.coordinator)

        viewModel.firebaseLoginResponse.observe(this, Observer {
            if(it){
                startActivity(Intent(this, MainScreen::class.java))
            } else{
                Toast.makeText(this, "Deu merda", Toast.LENGTH_LONG).show()
            }
        })
        firebaseAuth = FirebaseAuth.getInstance()
        confirm = findViewById(R.id.confirm_Button)
        confirm.setOnClickListener(View.OnClickListener {
            viewModel.validarCampo(user.text.toString()
                    ,password.text.toString()
                    ,passwordRepeat.text.toString())
        })
    }
}
